bycycle/cyclepoints/extrema.py

```diff
@@ -51,13 +54,20 @@ def find_extrema(sig, fs, f_range, boundary=None, first_extrema='peak', filter_k
     if filter_kwargs is None:
         filter_kwargs = {}
 
-    # Default boundary value as 1 cycle length of low cutoff frequency
-    if boundary is None:
-        boundary = int(np.ceil(fs / float(f_range[0])))
+    # Pad beginning of signal with zeros to prevent missing cyclepoints
+    if pad:
+        filt_len = compute_filter_length(fs, 'bandpass', f_range[0], f_range[1],
+                                         n_seconds=filter_kwargs.get('n_seconds', None),
+                                         n_cycles=filter_kwargs.get('n_cycles', 3))
+
+    else:
+        filt_len = 0
+
+    sig_pad = np.zeros(len(sig) + filt_len)
+    sig_pad[int(filt_len):] = sig
```

Comment:
So, I think right now, we only pad the beginning, to a full filter length?
Presumably we want to pad both ends of the signal?
Also, on one end, the padding only needs to be half a filter length. Once one gets half the filter length into the signal, the filter sees once full length data, and no longer has edge effects. 

So unless I'm missing something - the current length is good, but this should be updated to have the signal in the middle, with a half filter length buffer on either end.

Reply:
I initially had it in my head the edge artifacts only occurred at the beginning of signals, but this isn't true. Thanks for the catch! Updated as you suggest (round `filt_len` up to the nearest even number if needed, and pad half that length on each side).